debezium-core/src/main/java/io/debezium/rest/MetricsResource.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.rest;
+
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.management.AttributeNotFoundException;
+import javax.management.InstanceNotFoundException;
+import javax.management.IntrospectionException;
+import javax.management.MBeanException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.ReflectionException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.kafka.connect.connector.Connector;
+
+import io.debezium.rest.model.MetricsAttributes;
+import io.debezium.rest.model.MetricsDescriptor;
+
+public interface MetricsResource extends MetricsAttributes {
+    String CONNECTOR_METRICS_ENDPOINT = "/connectors/{connector-name}/metrics";
+
+    Connector getConnector();
+
+    MetricsDescriptor getMetrics(String connectorName, MBeanServer mBeanServer)
```

Comment:
I think, according to my other comment/s, the `MBeanServer` parameter can safely be removed. It is always passed to `#getMetrics()` but always only forwarded down to be used here in the MetricsResource class/interface/trait.